homeassistant/components/hardware/models.py

```diff
@@ -0,0 +1,22 @@
+"""Models for Hardware."""
+from __future__ import annotations
+
+from typing import Protocol, TypedDict
+
+from homeassistant.core import callback
+
+
+class BoardInfo(TypedDict):
+    """Board info type."""
+
+    image: str | None
+    name: str | None
+    url: str | None
+
+
+class HardwareProtocol(Protocol):
+    """Define the format of hardware platforms."""
+
+    @callback
+    def async_board_info(self) -> BoardInfo:
+        """Return board info."""
```

Comment:
The `HardwareProtocol` only supports providing basic information about a mainboard.
The idea is that this can be extended to provide other information for other hardware types.